src/daemon/ca/certauth.rs

```diff
@@ -964,6 +973,123 @@ impl CertAuth {
+    fn child_suspend_inactive(&self, child_handle: &ChildHandle) -> KrillResult<Vec<CaEvt>> {
+        let mut res = vec![];
+
+        let child = self.get_child(&child_handle)?;
+
+        if child.is_suspended() {
+            return Ok(res); // nothing to do, child is already suspended
+        }
+
+        let mut version = self.version;
+        let handle = &self.handle;
+
+        // Find all the certs in all RCs for this child and suspend them.
+        for (rcn, rc) in self.resources.iter() {
+            let certified_keys = child.issued(rcn);
+
+            if certified_keys.is_empty() {
+                continue;
+            }
+
+            let mut cert_updates = ChildCertificateUpdates::default();
+
+            for key in certified_keys {
+                if let Some(issued) = rc.issued(&key) {
+                    cert_updates.suspend(issued.clone());
+                }
+            }
+
+            res.push(CaEvtDet::child_certificates_updated(
+                handle,
+                version,
+                rcn.clone(),
+                cert_updates,
+            ));
+            version += 1;
+        }
+
+        info!("CA '{}' suspended inactive child '{}'", handle, child_handle);
+        res.push(CaEvtDet::child_suspended(handle, version, child_handle.clone()));
+
+        Ok(res)
+    }
+
+    // Unsuspend a child. The intention is that this is called automatically when
+    // a suspended (inactive) child CA is seen to contact this parent again.
+    //
+    // When a child is unsuspended we need to:
+    // - mark it as unsuspended
+    // - republish suspended certificates for it, which
+    //    - will not expire for another day
+    //    - do not exceed the current resource entitlements of the CA
+    // - other suspended certificates will just be removed.
+    //
+    // Then the child may or may not request new certificates as it sees fit.
+    // I.e. the unsuspend should be done before the child gets an answer to its
+    // RFC 6492 list request.
+    fn child_unsuspend(&self, child_handle: &ChildHandle) -> KrillResult<Vec<CaEvt>> {
+        let mut res = vec![];
+
+        let child = self.get_child(&child_handle)?;
+
+        if !child.is_suspended() {
+            return Ok(res); // nothing to do, child is not suspended
+        }
+
+        let mut version = self.version;
+        let handle = &self.handle;
+
+        // Find all the certs in all RCs for this child and suspend them.
+        for (rcn, rc) in self.resources.iter() {
+            let certified_keys = child.issued(rcn);
+
+            if certified_keys.is_empty() {
+                continue;
+            }
+
+            let mut cert_updates = ChildCertificateUpdates::default();
+
+            for key in certified_keys {
+                if let Some(issued) = rc.suspended(&key) {
+                    // check that the cert is actually not expired or about to expire and not overclaiming
```

Comment:
So, some certs will be republished but others that are valid but nearing (within a day of) expiration will not be. Is it okay to have this "grey" situation where some valid certs are published but others are not? Will this not be confusing?